crates/uv/src/commands/toolchain/install.rs

```diff
@@ -27,88 +29,129 @@ pub(crate) async fn install(
         warn_user!("`uv toolchain install` is experimental and may change without warning.");
     }
 
+    let start = std::time::Instant::now();
+
     let toolchains = InstalledToolchains::from_settings()?.init()?;
     let toolchain_dir = toolchains.root();
 
-    let request = if let Some(target) = target {
-        let request = ToolchainRequest::parse(&target);
-        match request {
-            ToolchainRequest::Any => (),
-            ToolchainRequest::Directory(_)
-            | ToolchainRequest::ExecutableName(_)
-            | ToolchainRequest::File(_) => {
-                writeln!(printer.stderr(), "Invalid toolchain request '{target}'")?;
-                return Ok(ExitStatus::Failure);
-            }
-            _ => {
-                writeln!(printer.stderr(), "Looking for {request}")?;
-            }
-        }
-        request
+    let requests = if targets.is_empty() {
+        vec![PythonDownloadRequest::default()]
     } else {
-        ToolchainRequest::default()
+        targets
+            .iter()
+            .map(|target| parse_target(target, printer))
+            .collect::<Result<_>>()?
     };
 
-    if let Some(toolchain) = toolchains
-        .find_all()?
-        .find(|toolchain| toolchain.satisfies(&request))
-    {
-        writeln!(
-            printer.stderr(),
-            "Found installed toolchain '{}'",
-            toolchain.key()
-        )?;
+    let installed_toolchains: Vec<_> = toolchains.find_all()?.collect();
+    let mut unfilled_requests = Vec::new();
+    for request in requests {
+        if let Some(toolchain) = installed_toolchains
+            .iter()
+            .find(|toolchain| request.satisfied_by_key(toolchain.key()))
+        {
+            writeln!(
+                printer.stderr(),
+                "Found installed toolchain '{}' that satisfies {request}",
+                toolchain.key()
+            )?;
+            if force {
+                unfilled_requests.push(request);
+            }
+        } else {
+            unfilled_requests.push(request);
+        }
+    }
 
-        if force {
-            writeln!(printer.stderr(), "Forcing reinstallation...")?;
+    if unfilled_requests.is_empty() {
+        if targets.is_empty() {
+            writeln!(
+                printer.stderr(),
+                "A toolchain is already installed. Use `uv toolchain install <request>` to install a specific toolchain.",
```

Comment:
I think the idea here is that if this message is printed, then I think it's guaranteed that some other "Found installed toolchain ..." message is printed above it, right?

Reply:
Also, should this say, "all requested toolchains are already installed"? Instead of just saying only a single toolchain is already installed?

Reply:
Yeah that's the idea. The output is actually pretty reasonable with both messages, one is listing discovered toolchains and the other summarizes the result of the whole operation.

We only say this when the targets are empty, which means we're looking for _any_ managed toolchain. This is a special case added in https://github.com/astral-sh/uv/pull/4248